dev-haskell/hnix/hnix-0.17.0.ebuild

```
@@ -0,0 +1,98 @@
+	>=dev-haskell/hnix-store-readonly-0.1.0:=[profile?]
+	>=dev-haskell/http-client-tls-0.3.5:=[profile?] <dev-haskell/http-client-tls-0.4:=[profile?]
+	>=dev-haskell/http-types-0.12.2:=[profile?] <dev-haskell/http-types-0.13:=[profile?]
+	>=dev-haskell/lens-family-1.2.2:=[profile?] <dev-haskell/lens-family-2.2:=[profile?]
+	>=dev-haskell/lens-family-core-1.2.2:=[profile?] <dev-haskell/lens-family-core-2.2:=[profile?]
+	>=dev-haskell/lens-family-th-0.5.0:=[profile?] <dev-haskell/lens-family-th-0.6:=[profile?]
+	>=dev-haskell/megaparsec-7.0:=[profile?] <dev-haskell/megaparsec-9.6:=[profile?]
+	>=dev-haskell/monad-control-1.0.2:=[profile?] <dev-haskell/monad-control-1.1:=[profile?]
+	>=dev-haskell/monadlist-0.0.2:=[profile?] <dev-haskell/monadlist-0.1:=[profile?]
+	>=dev-haskell/neat-interpolation-0.4:=[profile?] <dev-haskell/neat-interpolation-0.6:=[profile?]
+	>=dev-haskell/optparse-applicative-0.14.3:=[profile?] <dev-haskell/optparse-applicative-0.19:=[profile?]
+	>=dev-haskell/parser-combinators-1.0.1:=[profile?] <dev-haskell/parser-combinators-1.4:=[profile?]
+	>=dev-haskell/pretty-show-1.9.5:=[profile?] <dev-haskell/pretty-show-1.11:=[profile?]
+	>=dev-haskell/prettyprinter-1.7.0:=[profile?] <dev-haskell/prettyprinter-1.8:=[profile?]
+	>=dev-haskell/ref-tf-0.5:=[profile?] <dev-haskell/ref-tf-0.6:=[profile?]
+	>=dev-haskell/regex-tdfa-1.2.3:=[profile?] <dev-haskell/regex-tdfa-1.4:=[profile?]
+	>=dev-haskell/relude-1.0.0:=[profile?] <dev-haskell/relude-1.3:=[profile?]
+	>=dev-haskell/repline-0.4.0.0:=[profile?] <dev-haskell/repline-0.5:=[profile?]
+	>=dev-haskell/scientific-0.3.6:=[profile?] <dev-haskell/scientific-0.4:=[profile?]
+	>=dev-haskell/semialign-1.2:=[profile?] <dev-haskell/semialign-1.4:=[profile?]
+	>=dev-haskell/serialise-0.2.1:=[profile?] <dev-haskell/serialise-0.3:=[profile?]
+	>=dev-haskell/some-1.0.1:=[profile?] <dev-haskell/some-1.1:=[profile?]
+	>=dev-haskell/split-0.2.3:=[profile?] <dev-haskell/split-0.3:=[profile?]
+	>=dev-haskell/syb-0.7:=[profile?] <dev-haskell/syb-0.8:=[profile?]
+	>=dev-haskell/text-1.2.3:=[profile?] <dev-haskell/text-2.2:=[profile?]
+	>=dev-haskell/th-lift-instances-0.1:=[profile?] <dev-haskell/th-lift-instances-0.2:=[profile?]
+	>=dev-haskell/these-1.0.1:=[profile?] <dev-haskell/these-1.3:=[profile?]
+	>=dev-haskell/transformers-base-0.4.5:=[profile?] <dev-haskell/transformers-base-0.5:=[profile?]
+	>=dev-haskell/unix-compat-0.4.3:=[profile?] <dev-haskell/unix-compat-0.8:=[profile?]
+	>=dev-haskell/unordered-containers-0.2.14:=[profile?] <dev-haskell/unordered-containers-0.3:=[profile?]
+	>=dev-haskell/vector-0.12.0:=[profile?] <dev-haskell/vector-0.14:=[profile?]
+	>=dev-haskell/xml-1.3.14:=[profile?] <dev-haskell/xml-1.4:=[profile?]
+	>=dev-lang/ghc-9.0.2:=
+	>=dev-haskell/aeson-1.4.2:=[profile?] 
+	>=dev-haskell/http-client-0.5.14:=[profile?] <dev-haskell/http-client-0.8:=[profile?]
+	>=dev-haskell/logict-0.6.0:=[profile?] <dev-haskell/logict-0.9:=[profile?]
```

The `haskeline` and `repline` dependencies are only needed if `USE=executable` is enabled, so you should move those lines into a USE-conditional block.

```suggestion
	>=dev-haskell/logict-0.6.0:=[profile?] <dev-haskell/logict-0.9:=[profile?]
	executable? (
		>=dev-haskell/haskeline-0.8.0.0:=[profile?] <dev-haskell/haskeline-0.9:=[profile?]
		>=dev-haskell/repline-0.4.0.0:=[profile?] <dev-haskell/repline-0.5:=[profile?]
	)
```